pilot/pkg/networking/core/v1alpha3/tls.go

```diff
@@ -108,14 +112,39 @@ func buildOutboundTCPFilterChainOpts(node *model.Proxy, env *model.Environment,
 						destinationCIDRs: destinationCIDRs,
 						networkFilters:   buildOutboundNetworkFilters(node, clusterName, destinationIPAddress, listenPort),
 					})
+					hasTLSMatch = true
 				}
 			}
 		}
+	}
+
+	// HTTPS or TLS ports without associated virtual service will be treated as opaque TCP traffic.
+	if !hasTLSMatch {
+		clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, int(listenPort.Port))
+		out = append(out, &filterChainOpts{
+			destinationCIDRs: []string{destinationIPAddress},
+			networkFilters:   buildOutboundNetworkFilters(node, clusterName, destinationIPAddress, listenPort),
+		})
+	}
+
+	return out
```

Comment:
wont this cause two tcp proxies again? coz the other function is also adding a default route

Reply:
Only if there if a TCP and TLS port share the same port number, and the TLS port has no associated virtual services with a matching TLS block. That will be handled by the normal listener merging code.

In the normal case, we only add the default filter chain if the port is TCP or TLS. See:
https://github.com/istio/istio/pull/7394/files#diff-3ecbf67b650d37ef2b3af3c3bb19ba04R81
https://github.com/istio/istio/pull/7394/files#diff-3ecbf67b650d37ef2b3af3c3bb19ba04R137